src/apps/chifra/pkg/pinlib/chunk/errors.go

```diff
@@ -0,0 +1,42 @@
+package chunk
+
+import "fmt"
+
+// Errors that can happen when validating and saving a file
+
+type ErrSavingCorruptedDownload struct {
+	fileName string
+	expected int64
+	read     int64
+}
+
+func (err *ErrSavingCorruptedDownload) Error() string {
+	return fmt.Sprintf("file corrupted: %s (expected %d but got %d bytes)", err.fileName, err.expected, err.read)
+}
+
+type ErrSavingCannotDecompress struct {
+	fileName string
+	reason   error
+}
+
+func (err *ErrSavingCannotDecompress) Error() string {
```

Comment:
It's a nice idea, but because all of these strings need some data (like file names, etc) I guess we would have to put them into functions... and this would make the code a bit more complex I think.